### PR TITLE
Enh: Implement sortOrder & AdminMenu permissions

### LINKED ISSUE #21: Implement AdminMenu permission restriction
- [x] Implement `sortOrder` configurations
- [x] Implement AdminMenu permission restrictions
- [x] Development Testing
- [ ] Third-party testing
- [ ] Submit to marketplace

See https://github.com/humhub/humhub/issues/4695#issuecomment-742251519

### PR DESCRIPTION
**Pull request by ArchBlood**
Close #21

### CODE REVIEW

**Comment by ArchBlood** [comment]
@luke- can I get a code review? Everything looks alright on my end just need a third-party quick lookover

File: Events.php [hunk 1]
```diff
@@ -5,21 +5,31 @@
...
             'label' => Yii::t('WeatherModule.base', 'Weather Settings'),
             'url' => Url::toRoute('/weather/admin/index'),
-            'group' => 'settings',
-            'icon' => '<i class="fa fa-cloud"></i>',
+            'icon' => Icon::get('cloud'),
             'isActive' => Yii::$app->controller->module && Yii::$app->controller->module->id == 'weather' && Yii::$app->controller->id == 'admin',
-            'sortOrder' => 650
-        ]);
+            'sortOrder' => 650,
+            'isVisible' => true,
```
Comment (luke-) [inline Events.php:31]: You could also use:

'isVisible' => (!Yii::$app->user->can(ManageModules::class))

File: views/admin/index.php [hunk 2]
```diff
@@ -15,6 +15,7 @@
         <div class="form-group">
             <?= $form->field($model, 'serverUrl'); ?>
             <?= $form->field($model, 'location'); ?>
+            <?= $form->field($model, 'sortOrder')->textInput(['class' => 'form-control', 'placeholder' => Yii::t('WeatherModule.base', 'Select a sort order...'), 'disabled' => false])->label(true) ?>
```
Comment (luke-) [inline views/admin/index.php:18]: For 1.6+
```php
 <?= $form->field($model, 'sortOrder')->widget(humhub\modules\ui\form\widgets\SortOrderField::class) ?>
```

File: Module.php [hunk 3]
```diff
@@ -33,4 +34,13 @@ public function getLocation()
         }
         return $location;
     }
+
+    public function getOrder()
+    {
+        $sortOrder = $this->settings->get('sortOrder');
+        if (empty($sortOrder)) {
+            return '';
```
Comment (luke-) [inline Module.php:42]: return 100

**Review by luke- — APPROVED** [review]
Looks fine, some small remarks.

File: Events.php (hunk 1 above)
Comment (ArchBlood) [reply Events.php:31]: Using this removes it from view for admin as well, which has ManageModule, so I think we can continue using `'isVisible' => true`, or did you mean use `(Yii::$app->user->can(ManageModules::class))`? :thinking:

Comment (ArchBlood) [reply Events.php:31]: In any case I think setting this to `true` is good enough for now, thanks for the review! :smile:

**Pull request merged**